web-registry/src/lib/queries/react-query/ecocredit/queries.ecocredit.ts

```diff
@@ -0,0 +1,59 @@
+import { addDataToBatch, queryBatches } from 'lib/ecocredit/api';
+
+import {
+  ReactQueryAddDataToBatchesResponse,
+  ReactQueryAddDataToBatchParams,
+  ReactQueryBatchesProps,
+  ReactQueryBatchesResponse,
+} from './queries.ecocredit.types';
+
+/* getBatchesQuery */
+
+export const getBatchesQuery = ({
+  client,
+  request,
+  enabled,
+}: ReactQueryBatchesProps): ReactQueryBatchesResponse => ({
+  queryKey: [
+    'batches',
+    String(request.pagination?.offset ?? 0),
+    String(request.pagination?.limit ?? 0),
+  ],
+  queryFn: async () => {
+    if (client) {
+      const batches = await queryBatches({ client, request });
+
+      return batches;
+    }
+
+    return undefined;
+  },
```

Comment:
@flagrede wouldn't it be clearer by making a guard clause, to exit as soon as possible if the branch condition is not met?

```
queryFn: async () => {
  if (!client) return;                                                       // we can omit undefined here
  return await queryBatches({ client, request });         // no need to assign to a variable, we directly return the entire result
}
```